src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -712,10 +712,13 @@ private void Dispose(bool disposing)
             if (NetEventSource.Log.IsEnabled()) NetEventSource.Info(_state, $"{TraceId()} Stream disposing {disposing}");
 
             // If we haven't already shutdown gracefully (via a successful CloseAsync call), then force an abortive shutdown.
-            MsQuicApi.Api.ConnectionShutdownDelegate(
-                _state.Handle,
-                QUIC_CONNECTION_SHUTDOWN_FLAGS.SILENT,
-                0);
+            if (_state.Handle != null)
```

Comment:
Can we add a comment explaining that the handle can be null if the constructor throws?

Reply:
done.

Reply:
Also -- I wonder if we should be invoking any of the logic below at all if the handle is null. I guess it doesn't do any harm, right? It just seems like none of that applies in this case, and I worry we might introduce bugs in the future because we change that code and it no longer works in the case where the constructor failed.

Reply:
I was planning to take a look based on the review notes. Especially if we still observer any instability.